src/hotspot/share/gc/shared/referenceProcessor.cpp

```diff
@@ -367,44 +366,12 @@ size_t ReferenceProcessor::process_final_keep_alive_work(DiscoveredList& refs_li
     iter.next();
   }
   iter.complete_enqueue();
-  // Close the reachable set
-  complete_gc->do_void();
   refs_list.clear();
 
   assert(iter.removed() == 0, "This phase does not remove anything.");
   return iter.removed();
 }
 
-size_t ReferenceProcessor::process_phantom_refs_work(DiscoveredList&    refs_list,
-                                                     BoolObjectClosure* is_alive,
-                                                     OopClosure*        keep_alive,
-                                                     VoidClosure*       complete_gc) {
-  DiscoveredListIterator iter(refs_list, keep_alive, is_alive);
-  while (iter.has_next()) {
-    iter.load_ptrs(DEBUG_ONLY(!discovery_is_atomic() /* allow_null_referent */));
-
-    oop const referent = iter.referent();
-
-    if (referent == NULL || iter.is_referent_alive()) {
-      iter.make_referent_alive();
```

Comment:
The handling for `NULL` referents is (just a bit) different for phantom refs than for the others, but seems inconsequential.